### PR TITLE
Transaction Graph changes added

### PR DESCRIPTION
**Pull request by Harshi7016**

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import PersonDetail from '../components/user-info';
 import Filter from '../components/filter';
 import { Footer } from '../components/footer';
 import styles from '../styles/Home.module.css';
+import TransactionChart from '../components/transaction-chart/index.js';
```
Comment (swarajpure) [inline pages/index.js:11]: We can include just the component folder and not the index.js in it, following the convention from the existing codebase 😁

**Review by swarajpure — CHANGES_REQUESTED** [review]

File: pages/index.js (hunk 1 above)
Comment (Harshi7016) [reply pages/index.js]: Thanks for the review Swaraj, I have made the changes accordingly :)

**Comment by Harshi7016** [comment]
@Kratika0907 Please review the code and let me know for any changes, thanks for the help :)

File: components/transaction-chart/index.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import { Line } from 'react-chartjs-2';
+import { useState, useEffect } from 'react/cjs/react.development';
+import styles from './transaction-chart.module.css';
+const TransactionChart = (props) => {
+  const [chartData, setChartData] = useState({});
+
+  const chart = () => {
+    setChartData({
+      labels: [
+        '09/23/2020',
```
Comment (Kratika0907) [inline components/transaction-chart/index.js:11]: Please shift this data into mock that would be easy when we will refactor code base with actual api.

File: package.json [hunk 3]
```diff
@@ -14,12 +14,14 @@
   },
   "dependencies": {
     "@babel/plugin-transform-react-jsx": "^7.10.4",
+    "chart.js": "2.9.3",
```
Comment (Kratika0907) [inline package.json:17]: Please remove the un-necessary dependencies

File: components/transaction-chart/index.js (hunk 2 above)
Comment (Harshi7016) [reply components/transaction-chart/index.js:11]: Sure,I will definitely do that and will update

**Review by Kratika0907 — COMMENTED** [review]
Please make required changes, rest looks good to me.

File: package.json (hunk 3 above)
Comment (Harshi7016) [reply package.json:17]: So, I observed , react chart js was facing issue without chart js , both are dependent
Or required

**Comment by Harshi7016** [comment]
mockData for transaction graph is added

**Comment by Harshi7016** [comment]
@Kratika0907 , Now I guess I am clear with the way to create mockData, please let me know for any changes. I am kind of confident that I understood what is being expected now 
Sorry for so many changes

File: components/transaction-chart/index.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    transactionDates,
+    creditData,
+    debitData
+  } = props.transactionChartData;
+  const [chartData, setChartData] = useState({});
+  const chart = () => {
+    setChartData({
+      labels: transactionDates,
+      datasets: [
+        {
```
Comment (Kratika0907) [inline components/transaction-chart/index.js:16]: Here you can add a improvement, datasets require a array so I can write  a function which generate these objects and inside setChartData you can just call that function. This helps you in two ways : first my function is less cluttered and also add reusability of code.

**Review by Kratika0907 — CHANGES_REQUESTED** [review]

**Comment by Harshi7016** [comment]
Worked on creating mockData, this one looks promising

**Comment by Harshi7016** [comment]
The reason for the changes in the mockData occurred as I observed credit and debit are linked to timestamp and having separate timestamp for only debit or credit is creating irregularities in graph

**Review by Kratika0907 — APPROVED** [review]

**Review by sumitd94 — APPROVED** [review]

**Pull request merged**